lib/livebook/hubs.ex

```diff
@@ -54,18 +60,31 @@ defmodule Livebook.Hubs do
   def save_hub(struct) do
     attributes = struct |> Map.from_struct() |> Map.to_list()
     :ok = Storage.insert(@namespace, struct.id, attributes)
+    :ok = connect_hub(struct)
     :ok = broadcast_hubs_change()
+
     struct
   end
 
   @doc false
   def delete_hub(id) do
-    Storage.delete(@namespace, id)
+    if hub_exists?(id) do
+      hub = get_hub!(id)
+
+      if connected_hub = get_connected_hub(hub) do
+        GenServer.stop(connected_hub.pid)
```

Comment:
Should we assume it's a genserver, or have stop as part of the protocol? @josevalim

Reply:
Well, we're expecting a `child_spec` when  we calls `Provider.connect/1` from the protocol, so yeah, the connection should be a pid

Reply:
A pid for sure, but not technically doesn't need to be a genserver.

Reply:
I think it is fine to assume GenServer.stop for now. We can change later. :)